src/btree.c

```diff
@@ -1778,7 +1778,23 @@ btree_insert(cache              *cc,         // IN
    btree_node parent_node = root_node;
    btree_node child_node;
    child_node.addr = index_entry_child_addr(parent_entry);
-   debug_assert(allocator_page_valid(cache_get_allocator(cc), child_node.addr));
+#if SPLINTER_DEBUG
+   bool child_node_is_valid =
+      allocator_page_valid(cache_get_allocator(cc), child_node.addr);
+   if (!child_node_is_valid) {
+      btree_print_tree(Platform_default_log_handle,
+                       cc,
+                       (btree_config *)cfg,
+                       parent_node.addr,
+                       PAGE_TYPE_MEMTABLE);
```

Comment:
I had to deep-debug this area before I could id the fix for `mmap()` control flags. 

The new test case `test_one_insert_then_close_bug` in `splinterdb_forked_child_test.c` goes towards identifying this bug, where this debug code came very much in handy.